app/controllers/pledges_controller.rb

```diff
@@ -10,9 +10,11 @@ def new
   end
 
   def create
-    @pledge = @project.pledges.build(pledge_params)
-    if @pledge.save
+    @reward = Reward.find(params[:reward_id])
```

Comment:
sounds good! graceful degradation! Not sure of the difference between request.xhr and specifying json or javaScript.